src/pineko/scale_variations.py

```diff
@@ -0,0 +1,194 @@
+            # Set this subgrid inside the new grid
+            new_grid.set_subgrid(0, bin_index, lumi_index, extracted_subgrid)
+    return new_grid
+
+
+def create_grids(gridpath, order, nf):
+    """Create all the necessary scale variations grids for a certain starting grid."""
+    grid = pineappl.grid.Grid.read(gridpath)
+    grid_orders = [orde.as_tuple() for orde in grid.orders()]
+    first_nonzero_order = grid_orders[0]
+    m_value = first_nonzero_order[0]
+    deltaorder = order[0] - m_value
+    nec_orders = compute_orders_map(m_value, deltaorder)
+    grid_list = {}
+    for to_construct_order in nec_orders:
+        list_grid_order = []
+        for nec_order in nec_orders[to_construct_order]:
+            scalefactor = compute_scale_factor(
+                m_value, nec_order, to_construct_order, nf
+            )
+            list_grid_order.append(
+                create_svonly(grid, nec_order, to_construct_order, scalefactor)
+            )
+        grid_list[to_construct_order] = list_grid_order
+
+    return grid_list
+
+
+def write_sv_grids(gridpath, grid_list):
+    """Write the scale variations grids."""
+    base_name = gridpath.stem.split(".pineappl")[0]
+    final_part = ".pineappl.lz4"
+    grid_paths = []
+    for order in grid_list:
+        if len(grid_list[order]) > 1:
+            for grid in grid_list[order][1:]:
+                tmp_path = gridpath.parent / ("tmp" + final_part)
+                grid.raw.write_lz4(tmp_path)
+                grid_list[order][0].raw.merge_from_file(tmp_path)
+                os.remove(tmp_path)
+        new_grid_path = gridpath.parent / (
+            base_name + "_" + str(order[2]) + final_part
+        )  # order[2] is the ren_sv order
+        grid_paths.append(new_grid_path)
+        grid_list[order][0].raw.write_lz4(new_grid_path)
+    return grid_paths
+
+
+def merge_grids(gridpath, grid_list_path):
+    """Merge the scale variations grids."""
+    grid = pineappl.grid.Grid.read(gridpath)
+    base_name = gridpath.stem.split(".pineappl")[0]
+    new_path = gridpath.parent / (base_name + "_plusrensv.pineappl.lz4")
+    for grid_path in grid_list_path:
+        grid.raw.merge_from_file(grid_path)
+    grid.raw.write_lz4(new_path)
+
+
+def compute_ren_sv_grid(grid_path, max_as, nf):
+    """Generate renormalization scale variation terms for the given grid, according to the max_as.
+
+    Parameters
+    ----------
+    grid_pah : pathlib.Path()
+        pineappl grid path
+    max_as : int
+        max as order
+    nf : int
+        number of active flavors
+    """
+    # First let's check if the ren_sv are already there
+    grid = pineappl.grid.Grid.read(grid_path)
+    sv_as, sv_al = check.contains_ren(grid, max_as, max_al=0)
+    if sv_as:
+        rich.print(f"[green]Renormalization scale variations are already in the grid")
+    # Creating all the necessary grids
+    order = (max_as, 0, 0, 0)
+    grid_list = create_grids(grid_path, order, nf)
```

Comment:
Pass directly `max_as`:
```suggestion
    grid_list = create_grids(grid_path, max_as, nf)
```